pyproject.toml

```diff
@@ -1,10 +1,123 @@
+# Project Configuration
+################################################################################
+
+[project]
+name = "opentecr-schema"
+description = "contains the code for the schema of the openTECR database"
+authors = [
+    { name = "OpenTECR developers", email = "opentecr@googlegroups.com" }
+]
+maintainers = [
+    { name = "OpenTECR developers", email = "opentecr@googlegroups.com" }
+]
+license = "MIT"
+readme = {"file" = "README.md", "content-type" = "text/markdown"}
+requires-python = ">=3.8"
+# Please consult https://pypi.org/classifiers/ for a full list.
+classifiers = [
+    "Development Status :: 2 - Pre-Alpha",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Natural Language :: English",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Topic :: Scientific/Engineering :: Bio-Informatics"
+]
+# TODO: add keywords
+keywords = []
+dynamic = ["version"]
+dependencies = [
+    "depinfo ~=2.2",
+    "pydantic >= 2.0.0",
 ]
 
+[project.urls]
+Homepage = "https://github.com/rgiessman/opentecr-schema"
+Documentation = "https://opentecr-schema.readthedocs.io"
+"Source Code" = "https://github.com/rgiessman/opentecr-schema"
+"Bug Tracker" = "https://github.com/rgiessman/opentecr-schema/issues"
+Download = "https://pypi.org/project/opentecr-schema/#files"
+
+[project.optional-dependencies]
+dev = [
+    "mkdocs-material ~=9.1",
+    "mkdocstrings[python] ~=0.20",
+    "mkdocs-awesome-pages-plugin ~=2.8",
+    "pre-commit",
+    "tox ~=4.0"
+]
+
+################################################################################
+# Tool Configuration
+################################################################################
+
+[tool.hatch.build]
+only-packages = true
+
+[tool.hatch.build.targets.wheel]
+packages = ["src/opentecr"]
+
+[tool.hatch.build.hooks.vcs]
+version-file = "src/opentecr/_version.py"
+
+[tool.hatch.version]
+source = "vcs"
+
 [tool.black]
-python-version = ['py38']
+line-length = 88
+python-version = ["py311"]
+extend-exclude = '''
+(
+    _version.py
+)
+'''
+
+[tool.isort]
+profile = "black"
+lines_after_imports = 2
+lines_before_imports = 2
+known_first_party = "opentecr"
+skip = ["__init__.py"]
+
+[tool.pytest.ini_options]
+testpaths = ["tests"]
+markers = ["raises"]
+
+[tool.coverage.paths]
+source = [
+    "src/opentecr",
+    "*/site-packages/opentecr"
+]
+
+[tool.coverage.run]
+branch = true
+parallel = true
+
+[tool.coverage.report]
+exclude_lines = ["pragma: no cover"]
+precision = 2
+
+[tool.ruff]
+line-length = 88
+select = ["ALL"]
+ignore = [
+    "TD",  # ignore flake8-todo rules
+    "FIX",  # ignore flake8-fixme rules
+    "S101",  # allow assert statements 
```

Comment:
This is the only exception, that I'm not fully onboard with. If this is for test cases, there is a better way.

```toml
[tool.ruff.extend-per-file-ignores]
"test_*.py" = ["S101"]
```